### PR TITLE
 [ DO-1240 ] add and validate fields to users model

### PR DESCRIPTION
**Pull request by Alexandr8854**
add and validate fields to users model

### CODE REVIEW

**Comment by Alexandr8854** [comment]
@roman-tiukh I hope I did everything right

File: users/serializers.py [hunk 1]
```diff
@@ -15,11 +15,33 @@
 
 
 class DataOceanUserSerializer(serializers.ModelSerializer):
+
+    def validate(self, data):
+        """
+        validate fop and legal_entity fields
+        """
+        if data.get('persone_status') == 'fop' and 'legal_entity':
+            if data.get('iban') == '':
```
Comment (roman-tiukh) [inline users/serializers.py:24]: bad condition
```suggestion
            if data.get('iban'):
```

File: users/serializers.py [hunk 2]
```diff
@@ -15,11 +15,33 @@
 
 
 class DataOceanUserSerializer(serializers.ModelSerializer):
+
+    def validate(self, data):
+        """
+        validate fop and legal_entity fields
+        """
+        if data.get('persone_status') == 'fop' and 'legal_entity':
```
Comment (roman-tiukh) [inline users/serializers.py:23]: broken condition
```suggestion
        if data.get('persone_status') in ('fop', 'legal_entity'):
```

File: users/validators.py [hunk 3]
```diff
@@ -1,15 +1,22 @@
...
 
-
 two_in_row_validator = RegexValidator(
     regex=r"(.)\1{2,}",
     inverse_match=True,
     message=_("No more than 2 identical symbols in a row are allowed")
 )
+
+validate_iban = RegexValidator(regex='^UA\\d{8}[A-Z0-9]{19}$',
+                                   message = _('This field must contain 29 characters, numbers and Latin'))
```
Comment (roman-tiukh) [inline users/validators.py:18]:
```suggestion
                                   message = _('Wrong IBAN format'))
```

File: users/validators.py [hunk 4]
```diff
@@ -1,15 +1,22 @@
...
 )
 
-
 two_in_row_validator = RegexValidator(
     regex=r"(.)\1{2,}",
     inverse_match=True,
     message=_("No more than 2 identical symbols in a row are allowed")
 )
+
+validate_iban = RegexValidator(regex='^UA\\d{8}[A-Z0-9]{19}$',
```
Comment (roman-tiukh) [inline users/validators.py:17]: is this valid regex?

File: users/validators.py [hunk 5]
```diff
@@ -1,15 +1,22 @@
...
+                                   message = _('This field must contain 29 characters, numbers and Latin'))
+
+validate_edrpou = RegexValidator(regex='^[0-9]{8}$',
```
Comment (roman-tiukh) [inline users/validators.py:20]:
```suggestion
validate_edrpou = RegexValidator(regex='^\d{8}$',
```

File: users/serializers.py [hunk 6]
```diff
@@ -15,11 +15,33 @@
...
+              raise serializers.ValidationError(_("enter iban please"))
+        if data.get('persone_status') == 'legal_entity':
+            if data.get('edrpou') == '':
+                raise serializers.ValidationError(_("enter edrpou please"))
+        if data.get('persone_status') == 'legal_entity':
+            if data.get('name_company') == '':
+                raise serializers.ValidationError(_("enter name company please"))
+        if data.get('persone_status') == 'legal_entity':
+            if data.get('registration_address') == '':
+                raise serializers.ValidationError(_("enter registration address please"))
```
Comment (roman-tiukh) [inline users/serializers.py:34]: you 3 times check the same condition

File: users/serializers.py [hunk 7]
```diff
@@ -15,11 +15,33 @@
...
 class DataOceanUserSerializer(serializers.ModelSerializer):
+
+    def validate(self, data):
+        """
+        validate fop and legal_entity fields
+        """
+        if data.get('persone_status') == 'fop' and 'legal_entity':
+            if data.get('iban') == '':
+              raise serializers.ValidationError(_("enter iban please"))
+        if data.get('persone_status') == 'legal_entity':
```
Comment (roman-tiukh) [inline users/serializers.py:26]: Use constants from model 
```
DataOceanUser.LEGAL_ENTITY
```

File: users/validators.py (hunk 4 above)
Comment (roman-tiukh) [inline users/validators.py:17]:
```suggestion
iban_validator = RegexValidator(regex='^UA\\d{8}[A-Z0-9]{19}$',
```

File: users/validators.py (hunk 5 above)
Comment (roman-tiukh) [inline users/validators.py:20]:
```suggestion
edrpou_validator = RegexValidator(regex='^[0-9]{8}$',
```

File: users/serializers.py [hunk 8]
```diff
@@ -15,11 +15,33 @@
...
 
 class DataOceanUserSerializer(serializers.ModelSerializer):
+
+    def validate(self, data):
+        """
+        validate fop and legal_entity fields
+        """
+        if data.get('persone_status') == 'fop' and 'legal_entity':
+            if data.get('iban') == '':
+              raise serializers.ValidationError(_("enter iban please"))
```
Comment (roman-tiukh) [inline users/serializers.py:25]: you missed field name 
```suggestion
              raise serializers.ValidationError({'iban': _("This field is required for company or FOP"}))
```

File: users/models.py [hunk 9]
```diff
@@ -65,11 +66,23 @@ class DataOceanUser(AbstractUser):
...
+    name_company = models.CharField(max_length=150, default='', blank=True)
+    registration_address = models.CharField(max_length=150, default='', blank=True)
+    edrpou = models.CharField(max_length=8, default='', blank=True, validators=[validate_edrpou])
     # Permissions
-    datasets_admin = models.BooleanField(blank=True, default=False)
-    users_viewer = models.BooleanField(blank=True, default=False)
-    payment_system_admin = models.BooleanField(blank=True, default=False)
+    can_admin_registers = models.BooleanField(blank=True, default=False)
+    can_view_users = models.BooleanField(blank=True, default=False)
+    can_admin_payment_system = models.BooleanField(blank=True, default=False)
```
Comment (roman-tiukh) [inline users/models.py:85]: wrong changes

File: users/models.py [hunk 10]
```diff
@@ -65,11 +66,23 @@ class DataOceanUser(AbstractUser):
...
     )
-
+    INDIVIDUAL = 'individual'
+    FOP = 'fop'
+    LEGAL_ENTITY = 'legal_entity'
+    PERSONE_STATUSE = (
+        (INDIVIDUAL, _('Individual')),
+        (FOP, _('Fop')),
+        (LEGAL_ENTITY, _('Legal entity')),
+    )
```
Comment (roman-tiukh) [inline users/models.py:76]: move this to the top of model

File: users/models.py [hunk 11]
```diff
@@ -5,7 +5,8 @@
 from django.utils.translation import gettext_lazy as _
 from rest_framework.authtoken.models import Token
 from data_ocean.models import DataOceanModel
-from users.validators import name_symbols_validator, two_in_row_validator
+from django.core.exceptions import ValidationError
```
Comment (roman-tiukh) [inline users/models.py:8]: unused import

File: users/serializers.py [hunk 12]
```diff
@@ -15,11 +15,33 @@
 
 
 class DataOceanUserSerializer(serializers.ModelSerializer):
+
+    def validate(self, data):
+        """
+        validate fop and legal_entity fields
+        """
```
Comment (roman-tiukh) [inline users/serializers.py:22]: short docstring can be located in one line
```suggestion
        """ validate fop and legal_entity fields """
```

**Review by roman-tiukh — CHANGES_REQUESTED** [review]
migrations?

**Pull request closed without merging**